src/replication.c

```diff
@@ -1715,46 +1718,62 @@ void readSyncBulkPayload(connection *conn) {
      *
      * 2. Or when we are done reading from the socket to the RDB file, in
      *    such case we want just to read the RDB file in memory. */
-    serverLog(LL_NOTICE, "MASTER <-> REPLICA sync: Flushing old data");
 
     /* We need to stop any AOF rewriting child before flushing and parsing
      * the RDB, otherwise we'll create a copy-on-write disaster. */
     if (server.aof_state != AOF_OFF) stopAppendOnly();
 
-    /* When diskless RDB loading is used by replicas, it may be configured
-     * in order to save the current DB instead of throwing it away,
-     * so that we can restore it in case of failed transfer. */
     if (use_diskless_load &&
         server.repl_diskless_load == REPL_DISKLESS_LOAD_SWAPDB)
     {
-        /* Create a backup of server.db[] and initialize to empty
-         * dictionaries. */
-        diskless_load_backup = disklessLoadMakeBackup();
+        /* Initialize empty tempDb dictionaries. */
+        diskless_load_tempDb = disklessLoadInitTempDb();
+    }
+    else
+    {
+        // TODO: what's the effect of not calling this on REPL_DISKLESS_LOAD_SWAPDB now?
```

Comment:
The TODO was more questioning what is mentioned in the original comment:
/* OLD COMMENT: We call to emptyDb even in case of REPL_DISKLESS_LOAD_SWAPDB
        * (Where disklessLoadMakeBackup left server.db empty) because we
        * want to execute all the auxiliary logic of emptyDb (Namely,
        * fire module events) */
        
Mainly about modules, so touching this again after discussion on how to handle them.